config/pmd.xml

```diff
@@ -216,9 +235,13 @@
     <!-- too many false-positives -->
     <exclude name="BeanMembersShouldSerialize"/>
   </rule>
-  <rule ref="rulesets/java/logging-jakarta-commons.xml"/>
 
-  <rule ref="rulesets/java/logging-java.xml"/>
+  <rule ref="category/java/errorprone.xml/ProperLogger"/>
+  <rule ref="category/java/errorprone.xml/UseCorrectExceptionLogging"/>
+
```

Comment:
The rule `GuardLogStatement` is included in both `logging-jakarta-commons` and `logging-java`.
Due to bug in the PMD it can not be excluded. So, the references to `logging-jakarta-commons` and `logging-java` are replaced with their content.

Reply:
@pbludov What happens if they add a new rule to those packages? We will not be aware of them.

> Due to bug in the PMD it can not be excluded

Is there an issue number associated to this bug?

Reply:
This is a transitional PR. The following PRs remove all obsolete rulesets in favor of categories, so this problem will be gone.

Reply:
> What happens if they add a new rule to those packages? We will not be aware of them.

These rulesets are deprecated. New rules will be added to a category.
There is a new rule available only from the category: AvoidFileStream. See #5638 
The CS code violates this rule, but the config has no suppression for it.
So, if the PMD team will add new rules we will be not aware of them anyway.